Update Documentation For Private Github Pages (#15)
apple/swift-docc-plugin: Sources/SwiftDocCPluginDocumentation/SwiftDocCPlugin.docc/Publishing to GitHub Pages.md

    Your documentation site will be published at something like

    ```txt

>     https://<username>.github.io/<repository-name>/...

Reviewer: What do you think of leaving this with just the trailing slash and without the `...`s? 

```suggestion
    https://<username>.github.io/<repository-name>/
```

I could go either way here but my original intent was to match the URL that GitHub displays in the UI when you publish your site. For example:

<img width="799" alt="Screen Shot 2022-06-17 at 12 39 36 PM" src="https://user-images.githubusercontent.com/6750147/174391122-ce0df3cd-9550-497a-b141-525d09fb6b0b.png">

Author: Makes sense in that context, what I thought was, at the end we have the full url, so didn't want people to mistake this for the full url. Just an indication there might be something after that, we are going step by step, an invitation to read further. But yeah I too could go either way. If my concern is not valid then we can go ahead with what you suggested.

Reviewer: I think your point about people potentially mistaking this for the full URL is a really good one. Let's go with what you suggested and keep the `...`s in then. Thank you!

    ```

    and Swift-DocC needs to know about any base path after the `github.io` portion in order
    to correctly configure relative links. In the above case, that would be `<repository-name>`.
    

>     However, if you publish the page as a private github page, then the url structure will be different.

Reviewer: ```suggestion
    However, if you publish your website from a private GitHub repository, the URL structure is different and will not require a custom base path.
```

Author: Not exactly, we can publish a public page from a private repository. Only pages published as private will get this url structure. The repository visibility does not matter here if I am not mistaken.

Reviewer: Oh... I didn't know this was possible. You can publish a private GitHub pages site from a public repository?

Reviewer: Is this maybe a feature of GitHub enterprise? https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site

Author: Yes, it is a feature of enterprise. I discovered when I was contributing to my work, and was never a problem for personal repos. My original draft for this had text stating this actually.

Reviewer: Got it- okay maybe we should make things a little more general to future-proof things a bit?

```suggestion
    However, there are some configurations of GitHub Pages where this base URL path will not be required. For example, GitHub Enterprise Cloud has a feature that allows you to publish your GitHub Pages site privately. In situations like this, the URL structure will be different.
```

Author: ```suggestion
    However, if you publish your website, with visibility set as private, the URL structure is different and will not require a custom base path.
```
Looks good?

    
    Your documentation site will be published at something like
    
    ```txt

>     https://<unique-subdomain>.pages.github.io/...

Reviewer: ```suggestion
    https://<unique-subdomain>.pages.github.io/
```

    ```
    

>     As you can see, this is quite different from the public pages url. Each repo in your organization, will have its own subdomain, for hosting its private pages. Hence, the `[hosting-base-path]` argument is not needed for private pages, as it has its own subdomain.

Reviewer: What do you think of condensing this down a bit?

```suggestion
    Because websites published from private repositories have their own unique subdomain, your website is published directly at the root path and the `[hosting-base-path]` argument is not required.
```

Reviewer: And then maybe updating this to:

```suggestion
    Because these website have their own unique subdomain, your website is published directly at the root and the `[hosting-base-path]` argument is not needed. 
    
    > Tip: If your unsure what kind of GitHub Pages site you have, pay close attention to the URL listed on the GitHub Pages tab of your repository's settings after you've enabled the feature.
```

Author: ```suggestion
    Because these websites have their own unique subdomain, your website is published directly at the root and the `[hosting-base-path]` argument is not needed. 
    
    > Tip: If your unsure what kind of GitHub Pages site you have, pay close attention to the URL listed on the GitHub Pages tab of your repository's settings after you've enabled the feature.
```

|----------------------------|----------------------------------------------------------------------------------------------------------------|
| `[path-to-docs-directory]` | The path to the `/docs` directory at the root of the repository you configured for publishing to GitHub pages. |
| `[target-name]`            | The name of the Swift Package target you'd like to build documentation for.                                    |

> | `[hosting-base-path]`      | The base path your website will be hosted at. Most likely this will be the name of your GitHub repository. If private pages are being used, please refrain from using this option.

Reviewer: @denil-ct What do you think of leave this as-is for the common case but updating the more detailed documentation in step (1) above to clarify how this would work for a private GitHub repository?

Author: Yes, that seems much cleaner, and more space to explain the reasoning. Will make the change.

| `[target-name]`            | The name of the Swift Package target you'd like to build documentation for.                                    |

> | `[hosting-base-path]`      | The base path your website will be hosted at. Most likely this will be the name of your GitHub repository.     |

Reviewer: ```suggestion
| `[hosting-base-path]`      | The base path, if any, your website will be hosted at. Most likely this will be the name of your GitHub repository.     |
```

Once the push completes, the documentation site will be available at:

    https://<username>.github.io/<repository-name>/documentation/<target-name>
    

> If the documenation is published as a private page, then the documentation site will be available at:

Reviewer: ```suggestion
If the documentation is published from a private GitHub repository, then the documentation site will be available at:
```

Reviewer: And then, what do you think of this?

```suggestion
If your GitHub Pages site is published privately, then the documentation site will be available at:
```

Author: ```suggestion
If your GitHub Pages site is published privately, then the documentation will be available at:
```

Removed the site after documentation. Looks good?

Reviewer: Yeah I like that better. That's great- thank you!
